status/manager.go

```diff
@@ -53,6 +74,39 @@ func NewStatusManager() *Manager {
 	}
 }
 
+// IsSystemReady shows if the given sub-server ready to handle the a request for
+// the passed request URI. The first returned boolean is true if the system
+// is ready to handle the request. The second returned boolean is true if the
+// system has been disabled.
+func (s *Manager) IsSystemReady(name, req string) (bool, bool, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	server, ok := s.subServers[name]
+	if !ok {
+		return false, false, errors.New("a sub-server with " +
+			"name %s has not yet been registered")
+	}
+
+	if server.Disabled {
+		return false, true, nil
+	}
+
+	var isReady, handled bool
+
+	if server.isReadyOverride != nil {
+		isReady, handled = server.isReadyOverride(
+			req, server.manualStatus,
+		)
+	}
+
+	if !handled || server.Running {
+		return server.Running, false, nil
+	}
+
+	return isReady, false, nil
+}
```

Comment:
I find this a bit hard to read. 

How about:

```
        if server.disabled {
		return false, true, nil
	}

	// If this system has an override check set, then we first check that
	// to see if it overrides the "ready" status of the system.
	if server.isReadyOverride != nil {
		isReady, handled := server.isReadyOverride(
			req, server.manualStatus,
		)
		if handled {
			return isReady, false, nil
		}
	}

	return server.running, false, nil
```

Reply:
Thanks, I've updated to your version, with a slight modification to the first if check as we also need to ensure that the system isn't marked as `running` when we let the override function define if the request should be accepted or not. Hope that didn't ruin the readability too much :)!